tools/D.defs

```diff
@@ -46,28 +46,18 @@ ifeq ($(CONFIG_ARCH_SIM),y)
     DFLAGS += -mtriple=$(LLVM_ARCHTYPE)-apple-$(LLVM_ABITYPE)
   endif
 else ifeq ($(CONFIG_ARCH_RISCV),y)
-  # Traget triple is riscv[32|64][isa]-unknown-none-elf
+  # Target triple is riscv[32|64]-unknown-none-elf
+  DFLAGS += -mtriple=$(LLVM_ARCHTYPE)-unknown-none-elf
+  DFLAGS += -mattr=+m,+a,+f,+d,+c
 
-  D_ARCHTYPE = $(LLVM_ARCHTYPE)i
-  ifeq ($(CONFIG_ARCH_RV_ISA_M),y)
-    D_ARCHTYPE := $(D_ARCHTYPE)m
+  # Handle ABI and CPU
+  ifeq ($(CONFIG_ARCH_RV32),y)
+    DFLAGS += -mcpu=generic-rv32
+    DFLAGS += -mabi=ilp32d
```

Comment:
Only rv32gc/rv64gc supported by D?

Reply:
Sorry I'm not clear about rv32gc/rv64gc support in D. I was trying to restore the original intent for compiling D, which supports rv32gc only: https://github.com/apache/nuttx/commit/c6eea4ad8b80dcbc5c517ae2d3f6d59ba7db99d0#diff-052b9ec2c7c236a9a88a7cc78a9b4fc765e1741a3456a283174566665f74c995

```text
  # Traget triple is riscv[32|64][isa]-unknown-none-elf
  DFLAGS += -mtriple=$(LLVM_ARCHTYPE)$(ARCHRVISAM)$(ARCHRVISAA)$(ARCHRVISAF)$(ARCHRVISAD)$(ARCHRVISAC)-unknown-none-elf
  DFLAGS += -mattr=+m,+a,+f,+d,+c -mabi=ilp32d
  DFLAGS += -mcpu=generic-rv32
```

Hi @kassane: Wonder if you would like to expand the logic to handle other types of D Targets, like rv32imac? Or should we leave this for now? Thanks!

Reply:
The default is `riscv[32|64]-` no need `[isa]`.
```bash
$ ldc2 -mtriple=riscv32i -mcpu=help
Error: unable to get target for 'riscv32i', see -version and -mtriple.
$ ldc2 -mtriple=riscv32gc -mcpu=help
Error: unable to get target for 'riscv32gc', see -version and -mtriple.
```
It's strange because `ldmd2` (ldc2-wrapper, `dmd` flags compat) supports false positives on flags and doesn't report an error.

## Reference
- https://github.com/ldc-developers/ldc/pull/4322